### PR TITLE
editMode works

### PR DESCRIPTION
**Pull request by Aaron-Xiaoxingchen**
pls review my code!

Aaron

Thanks

### CODE REVIEW

File: src/watchlists/WatchList.jsx [hunk 1]
```diff
@@ -30,15 +41,24 @@ export default function watchlist({ item }) {
...
+              isEdit={isEdit}
+              isChecked={checkedItems[i]}
+              onSelect={() => updateChecked(i)}
+              ticker={stock.ticker}
+              currPrice={stock.currPrice}
+              dailyChange={stock.dailyChange}
+              volumn={stock.volumn}
+            />
+          ))}
+          {isEdit ? <EditorMode checkedItems={checkedItems} isEdit={isEdit} toggle={setIsEdit} />
```
Comment (FairyWind) [inline src/watchlists/WatchList.jsx]: It's better to have ? start from a new line.

File: src/watchlists/WatchListItem.jsx [hunk 2]
```diff
@@ -1,56 +1,77 @@
...
+        {isEdit ? null
+          : (
+            <>
+              <View style={styles.watchListItemColumn}>
+                <Text style={styles.watchListItemColumnText}>
+                  {currPrice}
+                </Text>
+              </View>
+              <View style={styles.watchListItemColumn}>
+                {dailyChange > 0 ? (
```
Comment (FairyWind) [inline src/watchlists/WatchListItem.jsx]: It's better to have ? and : start from new line

File: src/watchlists/styles.js [hunk 3]
```diff
@@ -99,4 +99,26 @@ export default StyleSheet.create({
...
+  editorModeRight: {
+    flexDirection: 'row',
+  },
+  editor: {
+    flex: 1,
+    alignSelf: 'flex-end',
+    padding: 10,
+  },
+  checkButton: {
+    // width: 20,
```
Comment (FairyWind) [inline src/watchlists/styles.js]: you can just delete these several lines

File: src/watchlists/Editor.jsx [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import { MaterialIcons } from '@expo/vector-icons';
+import { View, TouchableHighlight } from 'react-native';
+import PropTypes from 'prop-types';
+import styles from './styles';
+
+const Editor = ({ isEdit, toggle }) => (
```
Comment (ikarasz) [inline src/watchlists/Editor.jsx]: Lets call this component `EditButton`

File: src/watchlists/EditorMode.jsx [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import React from 'react';
+import { FontAwesome5 } from '@expo/vector-icons';
+import { View, Text, TouchableHighlight } from 'react-native';
+import PropTypes from 'prop-types';
+import styles from './styles';
+
+const editorMode = ({ checkedItems, isEdit, toggle }) => {
```
Comment (ikarasz) [inline src/watchlists/EditorMode.jsx]: Component names should be in PascalCase

Comment (ikarasz) [inline src/watchlists/EditorMode.jsx]: Lets call this component `EditorControls`, `EditControls`, `EditorFooter`, `EditorButtons`, etc...

File: src/watchlists/EditorMode.jsx [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+  const num = checkedItems.filter((checked) => checked).length;
+
+  return (
+    <View style={styles.editorMode}>
+      <TouchableHighlight style={styles.editorButton} underlayColor="#ffebeb">
+        <Text style={styles.editDeleteButton}>
+          <FontAwesome5 size={14} name="trash" color="#666" />
+          {' '}
+          {num}
```
Comment (ikarasz) [inline src/watchlists/EditorMode.jsx]: add some margin instead of an unnecessary character

File: src/watchlists/WatchList.jsx [hunk 7]
```diff
@@ -10,13 +10,24 @@ import WatchListItem from './WatchListItem';
 import chevron from '../../assets/icons/watchList/chevron.png';
 import Card from '../common/Card';
 import styles from './styles';
+import EditorMode from './EditorMode';
+import Editor from './Editor';
 
 export default function watchlist({ item }) {
   const [isOpen, setIsOpen] = useState(true);
+  const [isEdit, setIsEdit] = useState(false);
```
Comment (ikarasz) [inline src/watchlists/WatchList.jsx]: `isInEditMode`? as state, `toggleEditMode` as setter?

File: src/watchlists/WatchList.jsx [hunk 8]
```diff
@@ -10,13 +10,24 @@ import WatchListItem from './WatchListItem';
...
+  const [checkedItems, setCheckedItems] = useState(item.stocks.map(() => false));
 
   function handleClick() {
     setIsOpen(!isOpen);
   }
+
+  function updateChecked(i) {
```
Comment (ikarasz) [inline src/watchlists/WatchList.jsx]: find a better name

File: src/watchlists/WatchList.jsx [hunk 9]
```diff
@@ -30,15 +41,25 @@ export default function watchlist({ item }) {
...
-          dailyChange={stock.dailyChange}
-          volumn={stock.volumn}
-        />
-      )) : null}
+      {isOpen && (
+        <>
+          {item.stocks.map((stock, i) => (
+            <WatchListItem
+              key={stock.id}
+              isEdit={isEdit}
```
Comment (ikarasz) [inline src/watchlists/WatchList.jsx]: create a separated component for editable watchlist items, pay attention to the single responsibility. Components with more funcionlaity are much harder to test

File: src/watchlists/WatchList.jsx [hunk 10]
```diff
@@ -30,15 +41,25 @@ export default function watchlist({ item }) {
...
+              isChecked={checkedItems[i]}
+              onSelect={() => updateChecked(i)}
+              ticker={stock.ticker}
+              currPrice={stock.currPrice}
+              dailyChange={stock.dailyChange}
+              volumn={stock.volumn}
+            />
+          ))}
+          {isEdit
+            ? <EditorMode checkedItems={checkedItems} isEdit={isEdit} toggle={setIsEdit} />
```
Comment (ikarasz) [inline src/watchlists/WatchList.jsx]: I think these two could go to the same EditorFooter component
 - it renders the edit button if it not in edit mode
 - it renders the editor controls if it is in edit mode

File: src/watchlists/WatchListItem.jsx [hunk 11]
```diff
@@ -1,56 +1,83 @@
...
+  Text, View, Image, TouchableHighlight,
+} from 'react-native';
+import { MaterialIcons } from '@expo/vector-icons';
 import styles from './styles';
 import tradeIcon from '../../assets/icons/watchList/trade.png';
 import detailIcon from '../../assets/icons/watchList/detail.png';
 
-
 export default function watchListItem({
+  isEdit,
```
Comment (ikarasz) [inline src/watchlists/WatchListItem.jsx]: do not change this file at all with this pull request. Keep in mind the Open-closed principle

File: src/watchlists/EditModeWatchListItem.jsx [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+          name={isChecked ? 'check-box' : 'check-box-outline-blank'}
+        />
+        <View style={styles.watchListItemColumn}>
+          <Text style={[styles.watchListItemColumnText, { fontWeight: 'bold', color: '#566ed3' }]}>
+            {ticker}
+          </Text>
+        </View>
+      </View>
+    </TouchableHighlight>
+
```
Comment (ikarasz) [inline src/watchlists/EditModeWatchListItem.jsx]: unnecessary empty line

File: src/watchlists/EditModeWatchListItem.jsx [hunk 13]
```diff
@@ -0,0 +1,39 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {
+  Text, View, TouchableHighlight,
+} from 'react-native';
+import { MaterialIcons } from '@expo/vector-icons';
+import styles from './styles';
+
+export default function watchListItem({
```
Comment (ikarasz) [inline src/watchlists/EditModeWatchListItem.jsx]: use PascalCase for components

File: src/watchlists/WatchList.jsx (hunk 8 above)
Comment (ikarasz) [reply src/watchlists/WatchList.jsx]: still

**Pull request merged**